src/box/lua/console.lua

```diff
@@ -635,15 +655,27 @@ local function local_read(self)
             break
         end
         if delim == "" then
-            local lang = self.language
-            if not lang or lang == 'lua' then
-                -- stop once a complete Lua statement is entered
-                if local_check_lua(buf) then
+            local continuation_on
+            if self.remote ~= nil then
+                continuation_on = self.remote.continuation_on
+            else
+                continuation_on = self.continuation_on
+            end
+            local lang = self.language or 'lua'
+            -- Continue reading if input isn't a valid Lua statement or
+            -- expression or when it is syntactically complete but incorrect.
+            if (lang ~= 'lua' or local_check_lua(buf)) then
```

Comment:
> -- Continue reading if input isn't a valid Lua statement or
> -- expression <...>

The code does the opposite, not?

> <...> or when it is syntactically complete but incorrect.

Same.

Reply:
Rephrased:
```
              -- Continue reading if Lua input seems incomplete according to 
              -- internal heuristic.
```